runtime fixes, doc updates

justinmk opened the pull request.

justinmk committed:
vim-patch.sh: introduce `-V`

justinmk committed:
doc

vim-patch:8.0.1206: no autocmd for entering or leaving the command line
(commit a4f6cec7a31ff8dbfa089b9e22227afbeb951e9b)

NA patches:
vim-patch:8.0.0320: warning for unused variable with small build

justinmk committed:
vim-patch.sh: extract list_vimpatch_tokens()

Use streams instead of for-loop (20x speedup for list_vimpatch_tokens).

ZyX-I commented on runtime/doc/api.txt:
```diff
@@ -512,27 +512,27 @@ nvim_parse_expression({expr}, {flags}, {highlight})
-                    primary (debugging parser and providing debug
-                    information). "children": a list of nodes described in
-                    top/"ast". There always is zero, one or two children, key
-                    will not be present if node has no children. Maximum
```
Looks horrible here while being fine in online docs.

justinmk replied:
@ZyX-I Yes, need to enhance the doc-generation script somehow, or use doxygen "HTML". Not sure if doxygen discards the line break info.

justinmk committed:
vimpatch.lua: automate version.c

Invoke it like this:

    VIM_SOURCE_DIR=~/neovim/.vim-src/ nvim -i NONE -u NONE --headless +'luafile ./scripts/vimpatch.lua' +q

justinmk committed:
version.c: update
